SorrTask64: Refine Git workflow and practice doc

DanilYachmenev opened the pull request:
Task #64 

Follow-up from https://github.com/sorrentum/sorrentum/issues/64#issuecomment-1519166110

gpsaggese reviewed (commented):
Good work. Let's merge but pls
1) Add a line in the central Sorrentum README
2) Do a pass of the linter to reformat a bit the code

DanilYachmenev commented:
- Linter is applied, just moved the doc name on top of the TOC
- Updated README https://github.com/sorrentum/sorrentum/pull/228
- Updated Tecnical docs doc https://docs.google.com/document/d/14Ul5uWd7NU1zZ9ZLHp1TerLMN7MOfS5MS5bSX-2rBQ8/edit
- Original Gdoc is already striked out https://docs.google.com/document/d/1zahC8uDnFGYRSgkBrQRgg3W3ZmDjZZJj6yln6YeuHq4/edit#heading=h.gjamq2es185t
